src/elementPlugins/ErrorBoundary.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { Label } from 'semantic-ui-react'
+import { PLUGIN_ERRORS } from './pluginProvider'
+
+type Props = {
+  pluginCode: any
+  children: JSX.Element
+}
+
+type State = Readonly<{
+  hasError: boolean
+  errorMessage: string
+}>
```

Comment:
Can do, I wasn't sure what the practice was for classes (in projects like Grafana they seem to be in the same file).
I guess my argument for keeping them in classes, is because they won't be used anywhere else

Reply:
Yes, I had the same feeling. I keep all the props in the same file as the component. It seems less polluted without these here, like you suggested to move the function of a component to the bottom to keep it cleaner. 

I'll bring this up on our discussion today about Front-end, so we can decide what it's best for us.